src/commands/scanner/rule/add.ts

```diff
@@ -0,0 +1,78 @@
+import { flags, SfdxCommand } from '@salesforce/command';
+import { Messages, SfdxError } from '@salesforce/core';
+import { CustomRuleRegistrar } from '../../../lib/pmd/CustomRuleRegistrar';
+
+
+// Initialize Messages with the current plugin directory
+Messages.importMessagesDirectory(__dirname);
+
+// Load the specific messages for this file. Messages from @salesforce/command, @salesforce/core,
+// or any library that is using the messages framework can also be loaded this way.
+const messages = Messages.loadMessages('scanner', 'add');
+
+
+export default class Addcustom extends SfdxCommand {
+
+  public static description = messages.getMessage('commandDescription');
+
+  public static examples = [
+    `$ sfdx scanner:rule:add --language "apex" --path "/dir/to/jar/lib"
+        (todo: add sample output here)
+        
+        $ sfdx scanner:rule:add --language "apex" --path "/file/path/to/customrule.jar,/dir/to/jar/lib"
+        (todo: add sample output here)
+        `
+  ];
+
+  protected static flagsConfig = {
+    language: flags.string({ 
```

Comment:
This should be an array, so you can specify multiple languages at once.

Reply:
So far my changes take one language and its related paths at a time. Will revisit this in a bit.

Reply:
I'm not convinced that we should be taking in an array here.
Right now, Cataloger has the knowledge of which jars map to which languages, especially because PMD has a predictable jar file name based on language. Taking one language at a time and its corresponding paths will help maintain the language/path mapping. If we are taking an array of languages at once with multiple paths, the mapping would get further complicated.
User might as well use the add command once for each language. Or even multiple times for each language.

Reply:
I agree that for PMD, there's no reason for this to be an array, because each rule can only apply to one language. But long-term, our engine should be language agnostic, so you might want to say "these rules apply to all of these languages", without typing the command for each language separately. For now it can be a string, though.

Reply:
@rmohan20 what did you mean by, 'PMD has a predictable jar file name based on language'?  That is not true, is it?  Rules could be in any jar file name for any language, or not in a jar file at all (just a classes dir), right?

Reply:
The JARs that ship with PMD are all named 'pmd-[language]-[version].jar'. The cataloger uses that pattern to identify the JARs associated with the languages it's looking for. But with the customer rules, they'll be associating JARs with languages manually through the `add` command, so it doesn't matter terribly what they name them.

Reply:
@michaelhoefer Josh answered it perfectly in his last comment. Custom rules would continue to take JARs, libraries and directories.
@jfeingold35 Yeah, let's revisit this when we are language-agnostic. I'm expecting a few more changes around this command as well. For now, will keep it as a string and not a string array.